src/math/interpolator.cpp

```diff
@@ -474,3 +474,105 @@ void Interpolator::addInterpolated(Interpolator &source, Data1D &dest, double fa
     for (auto &&[x, y] : zip(dest.xAxis(), dest.values()))
         y += source.y(x) * factor;
 }
+
+// Add test function for benchmarks
+std::vector<double> Interpolator::y(std::vector<double> xs)
```

Comment:
We can pass in `xs` as a nice `const` reference to avoid copying it unnecessarily.
```suggestion
std::vector<double> Interpolator::y(const std::vector<double> &xs)
```